controllers/usersignup/usersignup_controller.go

```diff
@@ -591,6 +591,11 @@ func (r *Reconciler) checkIfMurAlreadyExists(reqLogger logr.Logger, config toolc
 			if err = r.ensureSpaceBinding(reqLogger, userSignup, mur, space); err != nil {
 				return true, err
 			}
+
+			// if the Space is not Ready, another reconcile will occur when space status is updated since this controller watches space (without a predicate)
+			if !r.isSpaceReady(reqLogger, space) {
+				return true, fmt.Errorf("space %s is not Ready", space.Name)
```

Comment:
We don't need to return an error - this is not an "error case". The controller should get automatically reconciled when the Space's status is modified.

Reply:
I mean - just keep the status of the UserSignup in an "incomplete" state (you can also introduce a new reason "Provisioning"), and just return here without an error. As soon as the Space.status is updated because it gets "ready" then the UserSignup will be triggered and the status will be updated to "complete".

Reply:
done in [5ea85c0](https://github.com/codeready-toolchain/host-operator/pull/670/commits/5ea85c0b2ec4c72293ee59beea0f8b8d3c8af09d)